File: demo-service/src/main/java/com/example/demo/data/Task.java
package com.example.demo.data;

import lombok.Builder;
import lombok.Value;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document("tasks")
@Value
@Builder(toBuilder = true)
public class Task {
    @Id
    String id;
    String title;
    String description;
    TaskStatus status;
    TaskPriority priority;
    Instant createdDate;
    Instant updatedDate;
}

File: demo-android/app/src/main/java/com/example/demo/data/model/Task.kt
package com.example.demo.data.model

data class Task(
    val id: String,
    val title: String,
    val description: String?,
    val status: TaskStatus,
    val priority: TaskPriority,
    val createdDate: String?,
    val updatedDate: String?
)

data class TaskRequest(
    val title: String,
    val description: String?,
    val status: TaskStatus,
    val priority: TaskPriority
)

data class ErrorResponse(
    val message: String?
)
